x-pack/plugins/security_solution_serverless/server/cloud_security/cloud_security_metering_task.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CNVM_POLICY_TEMPLATE,
+  CSPM_POLICY_TEMPLATE,
+  KSPM_POLICY_TEMPLATE,
+  LATEST_FINDINGS_INDEX_PATTERN,
+  LATEST_FINDINGS_RETENTION_POLICY,
+  LATEST_VULNERABILITIES_INDEX_PATTERN,
+  LATEST_VULNERABILITIES_RETENTION_POLICY,
+} from '@kbn/cloud-security-posture-plugin/common/constants';
+import type { UsageRecord } from '../types';
+
+import {
+  AGGREGATION_PRECISION_THRESHOLD,
+  CLOUD_SECURITY_TASK_TYPE,
+} from './cloud_security_metering';
+import { cloudSecurityMetringTaskProperties } from './cloud_security_metering_task_config';
+import type {
+  CloudSecurityMeteringCallbackInput,
+  PostureType,
+  ResourceCountAggregation,
+} from './types';
+
+const queryParams = {
+  [CSPM_POLICY_TEMPLATE]: {
+    index: LATEST_FINDINGS_INDEX_PATTERN,
+    timeRange: LATEST_FINDINGS_RETENTION_POLICY,
+  },
+  [KSPM_POLICY_TEMPLATE]: {
+    index: LATEST_FINDINGS_INDEX_PATTERN,
+    timeRange: LATEST_FINDINGS_RETENTION_POLICY,
+  },
+  [CNVM_POLICY_TEMPLATE]: {
+    index: LATEST_VULNERABILITIES_INDEX_PATTERN,
+    timeRange: LATEST_VULNERABILITIES_RETENTION_POLICY,
+  },
+};
+
+export const getCloudSecurityUsageRecord = async ({
+  esClient,
+  projectId,
+  logger,
+  taskId,
+  postureType,
+}: CloudSecurityMeteringCallbackInput): Promise<UsageRecord | undefined> => {
+  try {
+    if (!postureType) {
+      logger.error('posture type is missing');
+      return;
+    }
+
+    const response = await esClient.search<unknown, ResourceCountAggregation>(
+      getAggQueryByPostureType(postureType)
+    );
+
+    if (!response.aggregations) {
+      return;
+    }
+    const resourceCount = response.aggregations.unique_resources.value;
```

Comment:
As far as I understood the AGGREGATION_PRECISION_THRESHOLD is not in correlation to the resourceCount so I don't sure why to check and log this. 

In practice, we should choose a `precision_threshold` value that balances memory consumption and estimation accuracy based on your specific use case and available resources. If you set a higher precision_threshold, you will have a more accurate estimation for a larger number of unique values, but it will consume more memory. Conversely, a lower precision_threshold saves memory but may provide less accurate estimates for high-cardinality fields.

Reply:
done.

https://github.com/elastic/security-team/issues/7244